src/System Application/App/AppSource Gallery/src/AppSourceProductDetails.Page.al

```diff
@@ -0,0 +1,395 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.Apps.AppSource;
+
+using System.Apps;
+
+/// <summary>
+/// Single AppSource Product Details Page
+/// </summary>
+page 2516 "AppSource Product Details"
+{
+    PageType = Card;
+    ApplicationArea = All;
+    Editable = false;
+    Caption = 'App Overview';
+    DataCaptionExpression = AppSourceProductManager.GetStringValue(ProductObject, 'displayName');
+
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    layout
+    {
+        area(Content)
+        {
+            group(OfferGroup)
+            {
+                Caption = 'Offer';
+
+                field(Offer_UniqueID; UniqueProductID)
+                {
+                    Caption = 'Unique Product ID';
+                    ToolTip = 'Specifies the Unique Product ID';
+                    Visible = false;
+                }
+                field(Offer_ProductType; AppSourceProductManager.GetStringValue(ProductObject, 'productType'))
+                {
+                    Caption = 'Product Type';
+                    ToolTip = 'Specifies the Product Type';
+                    Visible = false;
+                }
+                field(Offer_DisplayName; AppSourceProductManager.GetStringValue(ProductObject, 'displayName'))
+                {
+                    Caption = 'Display Name';
+                    ToolTip = 'Specifies the Display Name';
+                }
+                field(Offer_PublisherID; AppSourceProductManager.GetStringValue(ProductObject, 'publisherId'))
+                {
+                    Caption = 'Publisher ID';
+                    ToolTip = 'Specifies the Publisher ID';
+                    Visible = false;
+                }
+                field(Offer_PublisherDisplayName; AppSourceProductManager.GetStringValue(ProductObject, 'publisherDisplayName'))
+                {
+                    Caption = 'Publisher Display Name';
+                    ToolTip = 'Specifies the Publisher Display Name';
+                }
+                field(Offer_PublisherType; AppSourceProductManager.GetStringValue(ProductObject, 'publisherType'))
+                {
+                    Caption = 'Publisher Type';
+                    ToolTip = 'Specifies the Publisher Type';
+                }
+                field(Offer_LastModifiedDateTime; AppSourceProductManager.GetStringValue(ProductObject, 'lastModifiedDateTime'))
+                {
+                    Caption = 'Last Modified Date Time';
+                    ToolTip = 'Specifies the Last Modified Date Time';
```

Comment:
Maybe these tooltips could explain a bit more, example this could say "Specifies when the App Source product was last updated." (I'm assuming that's what this means?

Reply:
Also if a function result is used as a field value, it disables filtering on it, if I'm not wrong. It would be better to build a temp record and fill it with values

Reply:
To address